src/elements/input/input.svelte

```diff
@@ -95,19 +95,30 @@ const handleInput = () => {
     return;
   }
 
-  if (type === 'number' && input.value.endsWith('.')) {
-    return;
+  if (type === 'number') {
+    // ensure input of chars that form valid numbers
+    let numString = '';
+    for (const c of input.value) {
+      if (/[\d+.e-]/.test(c)) {
+        numString += c;
+      }
+    }
+    input.value = value = numString;
+
+    // only send value if formatted as valid number
+    if (Number.isNaN(Number.parseFloat(value))) {
+      return;
+    }
+  } else {
+    input.value = value = input.value;
   }
-
-  value = input.value;
-
   internals.setFormValue(value);
 
   dispatch('input', { value });
 };
 
 const getDecimals = (value = '') => {
-  return Math.max(value.split('.').pop()?.length ?? 0, stepDecimalDigits);
```

Comment:
there was a bug where if value was a string representing an integer `"2"`, `value.split('.')` was `["2"]`, and then "value.split('.').pop()" was `"2"` which is a string of length `1`

Reply:
Nice catch!